src/quo/components/profile/collectible_list_item/view.cljs

```diff
@@ -67,108 +61,129 @@
            {:size   :paragraph-1
             :weight :semi-bold
             :style  style/card-detail-text}
-           collectible-name]]
+           collectible-name]])
 
-         :else
-         [:<>
-          [collection-avatar/view
-           {:size  :size-20
-            :image avatar-image-src}]
-          [rn/view {:style {:width 8}}]
-          [text/text
-           {:size            :paragraph-1
-            :weight          :semi-bold
-            :ellipsize-mode  :tail
-            :number-of-lines 1
-            :style           style/card-detail-text}
-           collectible-name]])])
+   [rn/view
+    {:style (style/avatar-container @loading?)}
+    (when-not @error?
+      [:<>
+       [collection-avatar/view
+        {:size        :size-20
+         :on-load-end #(reset! loading? false)
+         :on-error    #(reset! error? true)
+         :image       avatar-image-src}]
+       [rn/view {:style {:width 8}}]])
+    [text/text
+     {:size            :paragraph-1
+      :weight          :semi-bold
+      :ellipsize-mode  :tail
+      :number-of-lines 1
+      :style           style/card-detail-text}
+     collectible-name]]])
 
 (defn- card-view
-  [{:keys [avatar-image-src collectible-name community? counter
-           gradient-color-index image-src status]}]
+  [{:keys [avatar-image-src collectible-name community? counter avatar-error? avatar-loading?
+           gradient-color-index image-src image-loading? image-error? collectible-mime]}]
   (let [theme (quo.theme/use-theme-value)]
     [rn/view {:style (style/card-view-container theme)}
      [rn/view {:style {:aspect-ratio 1}}
       (cond
-        (= :loading status)
-        [loading-image
-         {:theme                theme
-          :gradient-color-index gradient-color-index}]
-
-        (= status :unsupported)
+        @image-error?
         [fallback-view
          {:theme theme
-          :label (i18n/label :t/unsupported-file)}]
+          :label (i18n/label :t/cant-fetch-info)}]
 
-        (= status :cant-fetch)
+        (not (utils/collectible-supported? collectible-mime))
         [fallback-view
          {:theme theme
-          :label (i18n/label :t/cant-fetch-info)}]
+          :label (i18n/label :t/unsupported-file)}]
 
-        :else
+        @image-loading?
+        [loading-image
+         {:theme                theme
+          :gradient-color-index gradient-color-index}])
+      (when (utils/collectible-supported? collectible-mime)
         [rn/view {:style {:aspect-ratio 1}}
          [rn/image
-          {:style  style/image
-           :source image-src}]])]
-     (when (and (not= status :loading) (not= status :cant-fetch) counter)
+          {:style       style/image
+           :on-load-end #(reset! image-loading? false)
```

Comment:
since this is new code, probably better to use react state since we are migrating towards that? 🤔

Reply:
Done